src/encoded/static/components/matrix_encore.js

```diff
@@ -0,0 +1,619 @@
+        <div className="general-search-entry">
+            <i className="icon icon-filter" />
+            <div className="searchform">
+                <input
+                    type="search"
+                    className="search-query"
+                    name="target-filter"
+                    id="target-filter"
+                    onChange={textChangeHandler}
+                    value={filterText}
+                />
+            </div>
+        </div>
+    </div>
+);
+
+TargetFilter.propTypes = {
+    /** Current value of filtering text */
+    filterText: PropTypes.string,
+    /** Called when the user changes a search input field */
+    textChangeHandler: PropTypes.func.isRequired,
+};
+
+TargetFilter.defaultProps = {
+    filterText: '',
+};
+
+
+/**
+ * Render the area above the facets and matrix content.
+ */
+const MatrixHeader = ({ context, targetFilterText, textChangeHandler }) => (
+    <div className="matrix-header">
+        <div className="matrix-header__title">
+            <div className="matrix-title-badge">
+                <h1>{context.title}</h1>
+                <MatrixBadges context={context} />
+            </div>
+            <div className="matrix-description">
+                <div className="matrix-description__text">
+                    The ENCORE project aims to study protein-RNA interactions by creating a map of RNA binding proteins (RBPs) encoded in the human genome and identifying the RNA elements that the RBPs bind to.
+                </div>
+            </div>
+        </div>
+        <div className="matrix-header__controls">
+            <div className="matrix-header__target-filter-controls">
+                <TargetFilter filterText={targetFilterText} textChangeHandler={textChangeHandler} />
+            </div>
+        </div>
+    </div>
+);
+
+MatrixHeader.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+    /** Current value of filtering text */
+    targetFilterText: PropTypes.string,
+    /** Called when the user changes a search input field */
+    textChangeHandler: PropTypes.func.isRequired,
+};
+
+MatrixHeader.defaultProps = {
+    targetFilterText: '',
+};
+
+
+/**
+ * Custom react hook to handle matrix scrolling.
+ * @param {node} Ref to scrolling matrix element
+ *
+ * @return {array} States and actions usable to this hook's clients
+ */
+const useMatrixScrollHandler = (scrollElement) => {
+    const [scrolledRight, setScrolledRight] = React.useState(true);
+
+    /**
+     * Show a scroll indicator depending on current scrolled position.
+     * @param {object} element DOM element to apply shading to
+     */
+    const handleScrollIndicator = React.useCallback((element) => {
+        // Have to use a "roughly equal to" test because of an MS Edge bug mentioned here:
+        // https://stackoverflow.com/questions/30900154/workaround-for-issue-with-ie-scrollwidth
```

Comment:
Which of the solutions did you use? It seems like you did parts of the selected solution and the solution by siphon. It is safe to assume developers won't go read it, so maybe explain why you have "scrollDiff <2"..?